File: services/payment-service/src/main/java/io/minishop/payment/web/dto/PaymentResponse.java
package io.minishop.payment.web.dto;

import io.minishop.payment.domain.Payment;
import io.minishop.payment.domain.PaymentStatus;

import java.math.BigDecimal;
import java.time.Instant;

public record PaymentResponse(
        Long id,
        Long orderId,
        Long userId,
        BigDecimal amount,
        PaymentStatus status,
        String externalRef,
        String failureReason,
        Integer attempts,
        Instant createdAt,
        Instant completedAt
) {
    public static PaymentResponse from(Payment p) {
        return new PaymentResponse(
                p.getId(), p.getOrderId(), p.getUserId(), p.getAmount(),
                p.getStatus(), p.getExternalRef(), p.getFailureReason(),
                p.getAttempts(), p.getCreatedAt(), p.getCompletedAt()
        );
    }
}

File: services/payment-service/src/main/kotlin/io/minishop/payment/web/dto/PaymentResponse.kt
package io.minishop.payment.web.dto

import io.minishop.payment.domain.Payment
import io.minishop.payment.domain.PaymentStatus
import java.math.BigDecimal
import java.time.Instant

@JvmRecord
data class PaymentResponse(
    val id: Long?,
    val orderId: Long?,
    val userId: Long?,
    val amount: BigDecimal?,
    val status: PaymentStatus?,
    val externalRef: String?,
    val failureReason: String?,
    val attempts: Int?,
    val createdAt: Instant?,
    val completedAt: Instant?,
) {
    companion object {
        @JvmStatic
        fun from(p: Payment): PaymentResponse = PaymentResponse(
            id = p.id,
            orderId = p.orderId,
            userId = p.userId,
            amount = p.amount,
            status = p.status,
            externalRef = p.externalRef,
            failureReason = p.failureReason,
            attempts = p.attempts,
            createdAt = p.createdAt,
            completedAt = p.completedAt,
        )
    }
}
